### DISCUSSION TITLE
Using onBlur and onChange with ComboBox in Mantine v7 and Integration with React Hook Form

### DISCUSSION THREAD

**Discussion by Sa-Daiki** [mantinedev/mantine#4918] [Q&A]
Hello Mantine team and community,

I recently migrated from Mantine v6 to v7 and have a couple of questions regarding the new ComboBox component:

Handling onBlur and onChange with ComboBox:
In Mantine v6, I was able to pass the onBlur event to the Select component. However, with the transition to using ComboBox in v7, I'm unsure how to pass the onBlur and onChange events. Could someone guide me on how to achieve this with the new ComboBox?

Integration with React Hook Form:
I'm also trying to integrate the ComboBox component with React Hook Form. Is it possible to use ComboBox with React Hook Form? If so, should I use the register method or the Controller component for this integration?

Any guidance or suggestions would be greatly appreciated. Thank you in advance!

**Comment by rtivital** [comment]
- Combobox itself does not handle any blur events, `onBlur` should be added to the input that is a target of Combobox.
- Yes, it is possible to use it with React Hook Form. Usually, you should go with Controller.
- Remember that Combobox is not an input, it is a helper to create custom inputs. You need to create a new input component with Combobox first and then integrate it with React Hook Form.

**Comment by enyelsequeira** [comment]
hey @Sa-Daiki where you able to come up with a solution?
